packages/nx/src/migrations/update-15-1-0/set-project-names.ts

```diff
@@ -5,13 +5,15 @@ import { readJson, writeJson } from '../../generators/utils/json';
 import { formatChangedFilesWithPrettierIfAvailable } from '../../generators/internal-utils/format-changed-files-with-prettier-if-available';
 import { retrieveProjectConfigurationPaths } from '../../project-graph/utils/retrieve-workspace-files';
 import { loadNxPlugins } from '../../project-graph/plugins/internal-api';
+import { shutdownPluginWorkers } from '../../project-graph/plugins/plugin-pool';
 
 export default async function (tree: Tree) {
   const nxJson = readNxJson(tree);
   const projectFiles = await retrieveProjectConfigurationPaths(
     tree.root,
     await loadNxPlugins(nxJson?.plugins)
   );
+  await shutdownPluginWorkers();
```

Comment:
do we need to call `shutdownPluginWorkers` in other migrations? Why do we need to call this anyway?